### ISSUE TITLE
[e2e-tests] Log stream interrupted

### ISSUE THREAD

**Issue by pebrc** [elastic/cloud-on-k8s#6693]
Test seems to run until `TestTelemetry/All_expected_Pods_should_eventually_be_ready`.  Test run is not marked as failed in buildkite.  To me it seems the output of the test just stops midway through as if the test was aborted. The go test runner does not seem to count these as failures and the whole test run is not marked as failed in buildkite.
However gotestsum seems to take the absence of an log entry with Action pass or fail as an implicit failure (which is fair enough)
Because the overall test did not fail we don’t have diagnostics

```
{"Time":"2023-04-16T21:28:43.6728121Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry"}
{"Time":"2023-04-16T21:28:43.672820647Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry","Output":"=== RUN   TestTelemetry\n"}
{"Time":"2023-04-16T21:28:43.672826993Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry","Output":"{\"log.level\":\"info\",\"@timestamp\":\"2023-04-16T21:28:43.672Z\",\"message\":\"Reusing image digest from cache\",\"service.version\":\"0.0.0-SNAPSHOT+00000000\",\"service.type\":\"eck\",\"ecs.version\":\"1.4.0\",\"image\":\"elasticsearch/elasticsearch\",\"tag\":\"8.8.0-SNAPSHOT\",\"digest\":\"sha256:cb7fdb68287abed16e99fbc3c09fb40323f45b415b7a3424c7330036fef33404\"}\n"}
{"Time":"2023-04-16T21:28:44.722941775Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry","Output":"I0416 21:28:44.722873   18322 request.go:690] Waited for 1.0464434s due to client-side throttling, not priority and fairness, request: GET:https://10.109.176.1:443/apis/kyverno.io/v1beta1?timeout=32s\n"}
{"Time":"2023-04-16T21:28:45.02564214Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/K8S_should_be_accessible"}
{"Time":"2023-04-16T21:28:45.025676288Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/K8S_should_be_accessible","Output":"=== RUN   TestTelemetry/K8S_should_be_accessible\n"}
{"Time":"2023-04-16T21:28:45.039315829Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/K8S_should_be_accessible","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.039345375Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Label_test_pods"}
{"Time":"2023-04-16T21:28:45.039349806Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Label_test_pods","Output":"=== RUN   TestTelemetry/Label_test_pods\n"}
{"Time":"2023-04-16T21:28:45.120038062Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Label_test_pods","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.120112627Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Elasticsearch_CRDs_should_exist"}
{"Time":"2023-04-16T21:28:45.120119745Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Elasticsearch_CRDs_should_exist","Output":"=== RUN   TestTelemetry/Elasticsearch_CRDs_should_exist\n"}
{"Time":"2023-04-16T21:28:45.124952973Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Elasticsearch_CRDs_should_exist","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.124982542Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Webhook_endpoint_should_not_be_empty"}
{"Time":"2023-04-16T21:28:45.124988381Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Webhook_endpoint_should_not_be_empty","Output":"=== RUN   TestTelemetry/Webhook_endpoint_should_not_be_empty\n"}
{"Time":"2023-04-16T21:28:45.128780373Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Webhook_endpoint_should_not_be_empty","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.128821477Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Remove_Elasticsearch_if_it_already_exists"}
{"Time":"2023-04-16T21:28:45.128828685Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Remove_Elasticsearch_if_it_already_exists","Output":"=== RUN   TestTelemetry/Remove_Elasticsearch_if_it_already_exists\n"}
{"Time":"2023-04-16T21:28:45.139732154Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Remove_Elasticsearch_if_it_already_exists","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.139764161Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/K8S_should_be_accessible#01"}
{"Time":"2023-04-16T21:28:45.139773471Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/K8S_should_be_accessible#01","Output":"=== RUN   TestTelemetry/K8S_should_be_accessible#01\n"}
{"Time":"2023-04-16T21:28:45.153392468Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/K8S_should_be_accessible#01","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.153417542Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Label_test_pods#01"}
{"Time":"2023-04-16T21:28:45.153423282Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Label_test_pods#01","Output":"=== RUN   TestTelemetry/Label_test_pods#01\n"}
{"Time":"2023-04-16T21:28:45.220331914Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Label_test_pods#01","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.220364129Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Kibana_CRDs_should_exist"}
{"Time":"2023-04-16T21:28:45.220369883Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Kibana_CRDs_should_exist","Output":"=== RUN   TestTelemetry/Kibana_CRDs_should_exist\n"}
{"Time":"2023-04-16T21:28:45.224040781Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Kibana_CRDs_should_exist","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.224074188Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Remove_Kibana_if_it_already_exists"}
{"Time":"2023-04-16T21:28:45.224081323Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Remove_Kibana_if_it_already_exists","Output":"=== RUN   TestTelemetry/Remove_Kibana_if_it_already_exists\n"}
{"Time":"2023-04-16T21:28:45.230637218Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Remove_Kibana_if_it_already_exists","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.230666619Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Creating_an_Elasticsearch_cluster_should_succeed"}
{"Time":"2023-04-16T21:28:45.230674268Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Creating_an_Elasticsearch_cluster_should_succeed","Output":"=== RUN   TestTelemetry/Creating_an_Elasticsearch_cluster_should_succeed\n"}
{"Time":"2023-04-16T21:28:45.244254313Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Creating_an_Elasticsearch_cluster_should_succeed","Output":"Retries (30m0s timeout): .{\"log.level\":\"info\",\"@timestamp\":\"2023-04-16T21:28:45.244Z\",\"log.logger\":\"KubeAPIWarningLogger\",\"message\":\"unknown field \\\"spec.http.service.metadata.creationTimestamp\\\"\",\"service.version\":\"0.0.0-SNAPSHOT+00000000\",\"service.type\":\"eck\",\"ecs.version\":\"1.4.0\"}\n"}
{"Time":"2023-04-16T21:28:45.244272556Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Creating_an_Elasticsearch_cluster_should_succeed","Output":"{\"log.level\":\"info\",\"@timestamp\":\"2023-04-16T21:28:45.244Z\",\"log.logger\":\"KubeAPIWarningLogger\",\"message\":\"unknown field \\\"spec.nodeSets[0].volumeClaimTemplates[0].metadata.creationTimestamp\\\"\",\"service.version\":\"0.0.0-SNAPSHOT+00000000\",\"service.type\":\"eck\",\"ecs.version\":\"1.4.0\"}\n"}
{"Time":"2023-04-16T21:28:45.244277861Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Creating_an_Elasticsearch_cluster_should_succeed","Output":"{\"log.level\":\"info\",\"@timestamp\":\"2023-04-16T21:28:45.244Z\",\"log.logger\":\"KubeAPIWarningLogger\",\"message\":\"unknown field \\\"spec.transport.service.metadata.creationTimestamp\\\"\",\"service.version\":\"0.0.0-SNAPSHOT+00000000\",\"service.type\":\"eck\",\"ecs.version\":\"1.4.0\"}\n"}
{"Time":"2023-04-16T21:28:45.244578092Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Creating_an_Elasticsearch_cluster_should_succeed","Output":"\n"}
{"Time":"2023-04-16T21:28:45.244654905Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Elasticsearch_cluster_should_be_created"}
{"Time":"2023-04-16T21:28:45.244697061Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Elasticsearch_cluster_should_be_created","Output":"=== RUN   TestTelemetry/Elasticsearch_cluster_should_be_created\n"}
{"Time":"2023-04-16T21:28:45.248590431Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Elasticsearch_cluster_should_be_created","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.24861065Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Creating_Kibana_should_succeed"}
{"Time":"2023-04-16T21:28:45.248616035Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Creating_Kibana_should_succeed","Output":"=== RUN   TestTelemetry/Creating_Kibana_should_succeed\n"}
{"Time":"2023-04-16T21:28:45.261244209Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Creating_Kibana_should_succeed","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.261262641Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Kibana_should_be_created"}
{"Time":"2023-04-16T21:28:45.261266344Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Kibana_should_be_created","Output":"=== RUN   TestTelemetry/Kibana_should_be_created\n"}
{"Time":"2023-04-16T21:28:45.267615601Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/Kibana_should_be_created","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:45.26765691Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/ES_HTTP_certificate_authority_should_be_set_and_deployed"}
{"Time":"2023-04-16T21:28:45.267662882Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/ES_HTTP_certificate_authority_should_be_set_and_deployed","Output":"=== RUN   TestTelemetry/ES_HTTP_certificate_authority_should_be_set_and_deployed\n"}
{"Time":"2023-04-16T21:28:48.292901916Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/ES_HTTP_certificate_authority_should_be_set_and_deployed","Output":"Retries (30m0s timeout): ..\n"}
{"Time":"2023-04-16T21:28:48.292944891Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/ES_transport_certificate_authority_should_be_set_and_deployed"}
{"Time":"2023-04-16T21:28:48.293008787Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/ES_transport_certificate_authority_should_be_set_and_deployed","Output":"=== RUN   TestTelemetry/ES_transport_certificate_authority_should_be_set_and_deployed\n"}
{"Time":"2023-04-16T21:28:48.296696626Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/ES_transport_certificate_authority_should_be_set_and_deployed","Output":"Retries (30m0s timeout): .\n"}
{"Time":"2023-04-16T21:28:48.296723822Z","Action":"run","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/All_expected_Pods_should_eventually_be_ready"}
{"Time":"2023-04-16T21:28:48.296729766Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/All_expected_Pods_should_eventually_be_ready","Output":"=== RUN   TestTelemetry/All_expected_Pods_should_eventually_be_ready\n"}
```

**Comment by thbkrkr** [comment]
Very strange. 10m30s after the last `TestTelemetry` log, the log stream ended and then resumed.
```
{"Time":"2023-04-16T21:28:48.296729766Z","Action":"output","Package":"github.com/elastic/cloud-on-k8s/v2/test/e2e/kb","Test":"TestTelemetry/All_expected_Pods_should_eventually_be_ready","Output":"=== RUN   TestTelemetry/All_expected_Pods_should_eventually_be_ready\n"}
{"log.level":"info","@timestamp":"2023-04-16T21:39:17.279Z","message":"Log stream ended","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","pod_name":"eck-e2e-72m9z-nmjvh"}
{"log.level":"info","@timestamp":"2023-04-16T21:39:17.279Z","message":"Streaming pod logs","service.version":"0.0.0-SNAPSHOT+00000000","service.type":"eck","ecs.version":"1.4.0","pod_name":"eck-e2e-72m9z-nmjvh"}
```

https://github.com/elastic/cloud-on-k8s/blob/ae88fe1662ed9f04fabc063edfdaaa4984ddb6d1/test/e2e/cmd/run/run.go#L693

**Comment by pebrc** [comment]
So this could just be a matter of an interruption of the log stream which would also explain why the go test runner does not count it as failure.

**Comment by pebrc** [comment]
I am not sure how to address these stream failures. We could look for a different way of transferring the logs e.g. retransferring all logs in the presence of stream failures (before the Job ends). Or as a simpler "work around" I was thinking we could look at the step outcome and if the build step has not failed suppress the failures for that build in the report.

**Comment by thbkrkr** [comment]
I've tried to figure out what's going on with no real success.

I've looked at the log stream end in different jobs (with and without false negatives) and there are always log stream interruptions. Sometimes it is not an issue, it starts again after 30sec-2min, we don't miss any log line. Sometimes it only starts again after 20-30 minutes and in this case we miss many log lines causing false negatives as described here.

It also seems that the main source of log stream interruptions is caused by what looks like a 4h proxy timeout between the bk agent (aws) and the test runner pod (gke). But when this kind of log stream interruption happens, the log stream is quickly restarted.

```
grep_stream() {
        grep -i stream $1 | grep -v Test_helper_streamTestJobOutput | grep '^{' \
                | sed "s/:[0-9][0-9].[0-9][0-9][0-9]Z//" \
                | jq '{t:.["@timestamp"],m:.message}' -c \
                | sort | uniq
}

# stack-8.6.2 (ok - despite a log stream interruption)

> grep_stream stack-8.6.2.log
{"t":"2023-05-21T13:26","m":"Streaming pod logs"}
{"t":"2023-05-21T14:08","m":"Log stream ended"} <<<<<<< +42min (explicit connection reset by peer)
{"t":"2023-05-21T14:08","m":"Streaming pod logs"}
{"t":"2023-05-21T18:08","m":"Log stream ended"} <<<<<<< +4h
{"t":"2023-05-21T18:08","m":"Streaming pod logs"}
{"t":"2023-05-21T20:36","m":"Log stream ended"} << end
{"t":"2023-05-21T20:36","m":"Log stream stopped"}
{"t":"2023-05-21T20:36","m":"Streaming pod logs"}
{"t":"2023-05-21T20:36","m":"Waiting for log stream to be over"}

## (ok)
2023-05-21T14:06:25.646Z TestSmoke/Elasticsearch_pods_should_eventually_be_removed
2023-05-21T14:08:26.082Z Log stream ended (read tcp 10.60.0.116:53740->34.65.32.192:443: read: connection reset by peer) <<<<<<< +0.4s (and we have a clear error here)

## (ok)
2023-05-21T18:07:48.772Z TestMutationRollingDownscaleCombination/Elasticsearch_pods_should_eventually_be_removed
2023-05-21T18:08:26.481Z Log stream ended <<<<<<< +38s

# kind-1.22.15 (ok - despite two log stream interruptions)

> grep_stream kind-1.22.15.log
{"t":"2023-05-21T13:20","m":"Streaming pod logs"}
{"t":"2023-05-21T17:20","m":"Log stream ended"}  <<<<<<< +4h
{"t":"2023-05-21T17:20","m":"Streaming pod logs"}
{"t":"2023-05-21T21:20","m":"Log stream ended"} <<<<<<< +4h
{"t":"2023-05-21T21:20","m":"Streaming pod logs"}
{"t":"2023-05-21T21:25","m":"Log stream ended"} << end
{"t":"2023-05-21T21:25","m":"Log stream stopped"}
{"t":"2023-05-21T21:25","m":"Streaming pod logs"}
{"t":"2023-05-21T21:25","m":"Waiting for log stream to be over"}

## (ok)
2023-05-21T17:20:11.118387464Z TestKillSingleNodeReusePV/All_expected_Pods_should_eventually_be_ready
2023-05-21T17:20:38.259Z Log stream ended <<<<<<< +27s

## (ok)
2023-05-21T21:20:13.929Z TestLogstashPipelineReload/Logstash_status_should_have_the_correct_status
2023-05-21T21:20:53.024Z Log stream ended <<<<<<< +40s

# stack-7-17-8 (ko)

> grep_stream stack-7-17-8.log
{"t":"2023-05-21T13:26","m":"Streaming pod logs"}
{"t":"2023-05-21T17:26","m":"Log stream ended"} <<<<<<< +4h
{"t":"2023-05-21T17:26","m":"Streaming pod logs"}
{"t":"2023-05-21T21:01","m":"Log stream ended"} <<<<<<< +3h35 << end
{"t":"2023-05-21T21:01","m":"Log stream stopped"}
{"t":"2023-05-21T21:01","m":"Streaming pod logs"}
{"t":"2023-05-21T21:01","m":"Waiting for log stream to be over"}

## (ok)
2023-05-21T17:24:14.792Z TestEnterpriseTrialLicense/Elasticsearch_pods_should_eventually_be_removed
2023-05-21T17:26:07.689Z Log stream ended <<<<<<< +1min53s
## (ko)
2023-05-21T20:41:25.325Z TestUpdateKibanaResources/Kibana_should_be_able_to_connect_to_Elasticsearch
2023-05-21T21:01:51.449Z Log stream ended <<<<<<< +20min25s

# stack-8.8.0-snap (ko)

> grep_stream stack-8.8.0-snap.log
{"t":"2023-05-21T13:25","m":"Streaming pod logs"}
{"t":"2023-05-21T14:14","m":"Log stream ended"}
{"t":"2023-05-21T14:14","m":"Streaming pod logs"}
{"t":"2023-05-21T18:14","m":"Log stream ended"} <<<<<<< +4h
{"t":"2023-05-21T18:14","m":"Streaming pod logs"}
{"t":"2023-05-21T21:38","m":"Log stream ended"} <<<<<<< +3h34 << end
{"t":"2023-05-21T21:38","m":"Log stream stopped"}
{"t":"2023-05-21T21:38","m":"Streaming pod logs"}
{"t":"2023-05-21T21:38","m":"Waiting for log stream to be over"}


## (ok)
2023-05-21T18:12:40.287Z TestUpdateESSecureSettings/Elasticsearch_secure_settings_should_eventually_be_set_in_all_nodes_keystore
2023-05-21T18:14:06.745Z Log stream ended <<<<<<< +1min53s
## (ko)
2023-05-21T21:02:20.234Z TestKBStackMonitoring/ES_transport_certificate_authority_should_be_set_and_deployed
2023-05-21T21:38:05.395Z Log stream ended <<<<<<< +35min45s
```

**Comment by thbkrkr** [comment]
My current conclusion is that the log stream file is no longer reliable and for lack of anything better we can look for another way to get the log as you suggested.

I'm wondering if we can not just run the script locally on the buildkite agent to write the log directly to it, which means no need to transfer the file anymore. I will test this.

**Comment by thbkrkr** [comment]
Closed by #6905.

**Comment by thbkrkr** [comment]
We now have false positives.

In the latest [build](https://buildkite.com/elastic/cloud-on-k8s-operator-nightly/builds/214), there was two e2e test failures and a more global issue with the resilience job:
- TestAPMServerVersionUpgradeToLatest8x ~ stack-8.0.1
- TestCrossNSAssociation ~ stack-8.9.0-SNAPSHOT

In jobs with test failures, strangely, the job pod is abruptly deleted, we never see it ready, and we don't get the result file. This is caused because when a test fails, we delete all pods in the e2e system namespace instead of the e2e managed namespaces.

https://github.com/elastic/cloud-on-k8s/blob/cf896c7029f70b7682e20115c3b233e9ba346224/test/e2e/test/utils.go#L219

I think that is what is the root cause of this issue that was causing the log stream interruption.

The problem with the resilience job is caused because in this case we deploy two jobs (chaos & e2e-tests) and since #6905, we stop the jobs as soon as we see that a pod is ready. We need to stop only after the e2e-tests job pod is ready and has downloaded the result file.

https://github.com/elastic/cloud-on-k8s/blob/cf896c7029f70b7682e20115c3b233e9ba346224/test/e2e/cmd/run/job_manager.go#L151

We also never run the diagnostic after a failure because we test if `CI == true` but this code is run from `go test` (`run.sh`) in the job pod where `CI` is not set.

https://github.com/elastic/cloud-on-k8s/blob/cf896c7029f70b7682e20115c3b233e9ba346224/test/e2e/test/diagnostics.go#L37

We should also fail if we can't download a test result file to not have this kind of false positive.

**Comment by thbkrkr** [comment]
Closed by #6922.
